### PR TITLE
Don't pass configuration to SDK autoconfigure through system props

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java [hunk 1]
```diff
@@ -61,98 +77,186 @@ public static Config get() {
...
+  @Nullable
+  public Boolean getBoolean(String name) {
+    return getTypedProperty(name, Boolean::parseBoolean, null);
+  }
+
+  /**
+   * Returns a boolean-valued configuration property or {@code defaultValue} if a property with name
+   * {@code name} has not been configured.
    */
   public boolean getBooleanProperty(String name, boolean defaultValue) {
```
Comment (mateuszrzeszutek) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:109]: I'm planning to rename all `get*Property()` methods to `get*()` in the next PR - those shorter names seem a bit nicer, and they're consistent with the SDK `ConfigProperties` interface too.

Comment (jkwatson) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:109]: is the plan to have this implement `ConfigProperties` then?

Comment (mateuszrzeszutek) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:109]: Unfortunately that won't be possible with the current shape of this repo: `instrumentation-api` only uses OTel API, not the SDK, so we can't directly implement the `ConfigProperties` interface here.

Comment (jkwatson) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java:109]: ok then. makes sense!

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java [hunk 2]
```diff
@@ -70,28 +69,6 @@ public static synchronized void installAgentTracer(Config config) {
...
-            .getAllProperties();
-
-    allProperties.forEach(
-        (key, value) -> {
-          if (!environmentProperties.containsKey(key)
-              && key.startsWith("otel.")
-              && !key.startsWith("otel.instrumentation")) {
-            System.setProperty(key, value);
-          }
-        });
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java:92]: 🎉

**Review by trask — APPROVED** [review]

**Pull request merged**